hspec-core/src/Test/Hspec/Core/Runner.hs

```diff
@@ -122,13 +123,17 @@ hspecWithResult conf spec = do
         seed = (fromJust . configQuickCheckSeed) c
         qcArgs = configQuickCheckArgs c
 
+    jobsSem <- case configMaxParallelJobs c of
+      Nothing -> return Nothing
```

Comment:
That makes a lot of sense to me.  There may be some workloads that are IO-bound rather than CPU-bound, so jobs in excess of getNumCapabilities could make sense.  However, the user could just select this manually if they need it.